File: 0804-unique-morse-code-words/0804-unique-morse-code-words.js
/**
 * @param {string[]} words
 * @return {number}
 */
var uniqueMorseRepresentations = function(words) {
    let morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."];
    let set = new Set();
    for(let word of words) {
        let str = "";
        for(let letter of word) {
            let index = letter.charCodeAt(0) - 97
            str += morse[index];
        }
        set.add(str);
    }
    return set.size;
};

File: 0804-unique-morse-code-words/0804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        unique_trans = set()
        
        for word in words:
            str = ""
            for letter in word:
                str += morse[ord(letter) - ord("a")]
            unique_trans.add(str)
            
        return len(unique_trans)